Tribler/Core/Modules/restapi/metadata_endpoint_base.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import absolute_import
+
+from twisted.web import resource
+
+from Tribler.Core.Modules.MetadataStore.serialization import CHANNEL_TORRENT, COLLECTION_NODE, REGULAR_TORRENT
+from Tribler.util import cast_to_unicode_utf8
+
+json2pony_columns = {
+    u'category': "tags",
+    u'id': "rowid",
+    u'name': "title",
+    u'size': "size",
+    u'infohash': "infohash",
+    u'date': "torrent_date",
+    u'updated': "torrent_date",
+    u'status': 'status',
+    u'torrents': 'num_entries',
+    u'votes': 'votes',
+    u'health': 'HEALTH',
+}
+
+# TODO: use the same representation for metatada nodes as in the database
+metadata_type_to_search_scope = {
+    '': [REGULAR_TORRENT, CHANNEL_TORRENT, COLLECTION_NODE],
+    "channel": [CHANNEL_TORRENT, COLLECTION_NODE],
+    "torrent": REGULAR_TORRENT,
+    str(CHANNEL_TORRENT): CHANNEL_TORRENT,
+    str(REGULAR_TORRENT): REGULAR_TORRENT,
+    str(COLLECTION_NODE): COLLECTION_NODE,
+}
+
+
+def convert_sort_param_to_pony_col(sort_param):
+    """
+    Convert an incoming sort parameter to a pony column in the database.
+    :return a string with the right column. None if there exists no value for the given key.
+    """
+
+    return json2pony_columns[sort_param] if sort_param in json2pony_columns else None
+
+
+class MetadataEndpointBase(resource.Resource):
```

Comment:
Could you rename this to `BaseMetadataEndpoint` ?

Reply:
I feel it is better to leave `Base` at the end to reflect the fact that this is *not* an endpoint for some object called `BaseMetadata`.